manager/allocator/networkallocator/networkallocator.go

```diff
@@ -284,7 +284,8 @@ func (na *NetworkAllocator) IsTaskAllocated(t *api.Task) bool {
 }
 
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
-func (na *NetworkAllocator) IsServiceAllocated(s *api.Service) bool {
+// init bool indicates if the func is called during allocator initialization stage.
+func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, init bool) bool {
```

Comment:
May I suggest using functional args instead? This will be easier to read and easier to expand in the future.

```go
type ServiceAllocationOpts struct {
        onInit bool
}

type ServiceAllocationFlag func(*ServiceAllocationOptions)

func OnInit(options *ServiceAllocationOptions) {
        options.onInit = true
}

func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags ...ServiceAllocationFlag) bool {
        var options ServiceAllocationOpts

        for _, flag := range flags {
                flag(&options)
        }
}
```